File: baekjoon/python/Q1.py
# 두 개 뽑아서 더하기

numbers = [2, 1, 3, 4, 1]

def solution(numbers):
    answer = []

    for i in range(len(numbers) - 1):
        for j in numbers[i + 1:]:
            answer.append(numbers[i] + j)
    return sorted(list(set(answer)))

# def solution(numbers):
#     answer = []

#     for i in range(0, len(numbers) - 1):
#         for j in numbers[i + 1:]:
#             s = numbers[i] + j
#             if not s in answer:
#                 answer.append(s)
#     answer.sort()
#     return answer

print(solution(numbers))

File: programmers/Q1.js
// 두 개 뽑아서 더하기

let numbers = [2, 1, 3, 4, 1];

function solution(numbers) {
  let temp = [];

  for (let i = 0; i < numbers.length; i++) {
    for (let j = i + 1; j < numbers.length; j++) {
      let addNumber = numbers[i] + numbers[j];
      temp.push(addNumber);
    }
  }

  const answer = [...new Set(temp)]; // spread
  return answer.sort((a, b) => a - b);
}

// function solution(numbers) {
//   let answer = [];

//   for (let i = 0; i < numbers.length; i++) {
//     for (let j = i + 1; j < numbers.length; j++) {
//       let addNumber = numbers[i] + numbers[j];
//       if (answer.indexOf(addNumber) === -1) {
//         answer.push(addNumber);
//       }
//     }
//   }
// //   console.log(answer.indexOf(10));
//   console.log(answer.sort((a, b) => a - b));
//   return answer.sort((a, b) => a - b);
// }

console.log(solution(numbers));
